sdk/python/kfp/compiler/_container_builder.py

```diff
@@ -47,6 +50,26 @@ def __init__(self, gcs_staging=None, namespace=None):
     from ._gcs_helper import GCSHelper
     GCSHelper.create_gcs_bucket_if_not_exist(gcs_bucket)
 
+    # Configure the GCR image tag
+    # Kubeflow jupyter notebook is injected with a variable as an ID
+    if gcr_image_tag is None:
+      KF_NOTEBOOK_ENV = 'NB_PREFIX'
+      nb_id = str(uuid.uuid4())
```

Comment:
I just found out that this case will never be used. 
The ContainerBuilder only uses the uuid when it is not in the KF jupyter notebook or it could not get the instance id from the GCE metadata service. In which case, get_project_id will also fail because it needs access to the GCE metadata service.
How about removing the uuid assignment to avoid confusion?

Reply:
sgtm. Maybe promoting user to provide the gcs and gcr path explicitly.

Reply:
SG